newa/__init__.py

```diff
@@ -273,6 +277,10 @@ def _test_match(s: str, pattern: str) -> bool:
     return bool(outcome == 'true')
 
 
+def get_url_basename(url: str) -> str:
+    return os.path.splitext(os.path.basename(urllib.parse.urlparse(url).path))[0]
```

Comment:
Is `splitext` still needed?

Reply:
basename will provide filename including an extension but I want to cut the extension off. I can see the function name is misleading in this context. Since I am going to use the final name as a default name for RP launch, I think I will move the `splitext` call to the respective place and let this function to return truly a basename. Will fix it right away.